addons/godot-next/references/tween_sequence.gd

```diff
@@ -0,0 +1,337 @@
+func set_autokill(autokill: bool):
+	__kill_when_finised = autokill
+
+func __add_tweener(tweener: Tweener):
+	if not __parallel:
+		__tweeners.append([])
+	__tweeners.back().append(tweener)
+	__parallel = false
+
+func __run_next_step() -> void:
+	assert(!__tweeners.empty(), "Sequence has no steps!")
+	var group := __tweeners[__current_step] as Array
+	for tweener in group:
+		tweener.__start(__tween)
+
+func __step_complete() -> void:
+	emit_signal("step_finished", __current_step)
+	__current_step += 1
+	
+	if __current_step == __tweeners.size():
+		__loops -= 1
+		if __loops == -1:
+			emit_signal("finished")
+			if __kill_when_finised:
+				kill()
+			else:
+				reset()
+		else:
+			emit_signal("loop_finished")
+			__current_step = 0
+			__run_next_step()
+	else:
+		__run_next_step()
+
+##Abstract class for all Tweeners.
+class Tweener:
+	extends Reference
+	
+	func __start(tween: Tween) -> void:
+		pass
+
+##Tweener for tweening properties.
+class PropertyTweener:
+	extends Tweener
+	
+	var __target: Object
+	var __property: NodePath
+	var __from
+	var __to
+	var __duration: float
+	var __trans: int
+	var __ease: int
+	
+	var __delay: float
+	var __continue := true
+	var __advance := false
+	
+	func _init(target: Object, property: NodePath, to_value, duration: float) -> void:
+		__target = target
+		__property = property
+		__from = __target.get(property)
+		__to = to_value
+		__duration = duration
+		__trans = Tween.TRANS_LINEAR
+		__ease = Tween.EASE_IN_OUT
+	
+	##Sets custom starting value for the tweener.
+	##By default, it starts from value at the start of this tweener.
+	func from(val) -> PropertyTweener:
+		__from = val
+		__continue = false
+		return self
+	
+	##Sets the starting value to the current value,
+	##i.e. value at the time of creating sequence.
+	func from_current() -> PropertyTweener:
+		__continue = false
+		return self
+	
+	##Sets transition type of this tweener, from Tween.TransitionType.
+	func set_trans(t: int) -> PropertyTweener:
+		__trans = t
+		return self
+	
+	##Sets ease type of this tweener, from Tween.EaseType.
+	func set_ease(e: int) -> PropertyTweener:
+		__ease = e
+		return self
+	
+	##Sets the delay after which this tweener will start.
+	func set_delay(d: float) -> PropertyTweener:
+		__delay = d
+		return self
+	
+	func __start(tween: Tween) -> void:
+		if __continue:
+			__from = __target.get(__property)
+		
+		if __advance:
+			tween.interpolate_property(__target, __property, __from, __from + __to, __duration, __trans, __ease, __delay)
+		else:
+			tween.interpolate_property(__target, __property, __from, __to, __duration, __trans, __ease, __delay)
+		tween.start()
+
+##Generic tweener for creating delays in sequence.
+class IntervalTweener:
+	extends Tweener
+	
+	var __time: float
+	
+	func _init(time: float) -> void:
+		__time = time
+	
+	func __start(tween: Tween) -> void:
+		tween.interpolate_callback(self, __time, "__")
+		tween.start()
+	
+	func __():
+		pass
```

Comment:
I wonder what does this method? Implementation-specific?

Reply:
Nothing. I could do interval with SceneTreeTimer, but since it's about Tweens, I did an empty callback tween that creates artificial interval. I just needed a method that does nothing.